hw/scripts/common.sh

```diff
@@ -88,6 +88,14 @@ detect_programmer() {
         # extract the VID:PID list for connected USB devices
         USB_DEV=$(lsusb | cut -f6 -d' ')
 
+    elif [ $(which system_profiler) ] ; then
+
+        # extract the VID:PID list for connected USB devices on OSX
+        USB_DEV=$(system_profiler SPUSBDataType 2>/dev/null | awk '/^[ \t]+Product ID:/ { PID=$3 } /^[ \t]+Vendor ID:/ { printf("%s:%s\n", PID, $3) }')
```

Comment:
Thanks for the contribution, however, I do think this statement should be reduced into smaller consecutive statements to make it consistent with rest of the script.

Reply:
For me this line looks perfectly OK.
I don't see anything that could be split.
Same approach is use few line above where `lsusb` is piped to `cut` and here `system_profiler` is piped to `awk` in very nice way including error redirection to **/dev/null**

I don't have OSX to check for system_profiler output but awk part is very clear and there is no single character I would written better.
@vrahane please reconsider

Reply:
@kasjer  Functionality wise, might be fine but just from a style perspective it is a very long line. In the entire script people have taken efforts to reduce the length of the statements. So, I would expect the same here. For example:
https://linuxhint.com/split-long-bash-command/

Reply:
@vrahane if your comment was about reformatting I would not object (@cqexbesd surely knows how to use `\`), but **smaller consecutive statements** did not apply here in my opinion.

Reply:
Either reformatting or using smaller consecutive statements, both are fine as long as it’s smaller. 151 characters in one line is not really a good practice.

Reply:
Reformatted, and a bug fix added.